elobot.py

```diff
@@ -54,245 +30,233 @@ def get_channel_id(self, channel_name):
+            name = self.slack_client.get_name(user_handle)
+            table.append([name, get_elo(user_handle), get_wins(user_handle), get_losses(user_handle), streak_text])
 
         self.talk('```' + tabulate(table, headers=['Name', 'ELO', 'Wins', 'Losses', 'Streak']) + '```')
 
     def print_unconfirmed(self):
         table = []
 
-        for match in Match.select().where(Match.pending == True).order_by(Match.played.desc()).limit(25):
-            match_played_utc = match.played.replace(tzinfo=from_zone)
-            match_played_pst = match_played_utc.astimezone(to_zone)
+        # TODO: There must be a cleaner way to do this line
+        for match in it.islice(filter(lambda m: m.pending, Match.select().order_by(Match.datetime.desc())), 0, 25):
+            match_datetime_utc = match.datetime.replace(tzinfo=from_zone)
+            match_datetime_pst = match_datetime_utc.astimezone(to_zone)
             table.append([
                 match.id,
-                self.slack_client.get_name(match.loser_handle),
-                self.slack_client.get_name(match.winner_handle),
-                '{} - {}'.format(match.winner_score, match.loser_score),
-                match_played_pst.strftime('%m/%d/%y %I:%M %p')
+                ' '.join(map(lambda p: self.slack_client.get_name(p.handle) + ('*' if p.pending else ''), match.winners)),
+                ' '.join(map(lambda p: self.slack_client.get_name(p.handle) + ('*' if p.pending else ''), match.losers)),
+                '{} - {}'.format(match.winners_score, match.losers_score),
+                match_datetime_pst.strftime('%m/%d/%y %I:%M %p')
             ])
 
-        self.talk('```' + tabulate(table, headers=['Match', 'Needs to Confirm', 'Opponent', 'Score', 'Date']) + '```')
-
-    def get_win_streak(self, player_slack_id):
-        win_streak = 0
-        matches = Match.select().where(Match.pending == False, (player_slack_id == Match.winner_handle) | (player_slack_id == Match.loser_handle)).order_by(Match.played.desc())
-        return len(list(takewhile(lambda m: m.winner_handle == player_slack_id, matches)))
+        self.talk('```* Needs to confirm\n' + tabulate(table, headers=['Match', 'Winning team', 'Losing team', 'Score', 'Date']) + '```')
 
 if __name__ == '__main__':
     with open('config.json') as config_data:
         config = json.load(config_data)
 
+    if config.get('debug', False):
+        print('Warning: backdoor enabled!')
+        BACKDOOR_ENABLED = True
+
     slack_client = SlackClient(config['slack_token'])
     db.connect()
-    Match.create_table()
-    EloBot(
+    create_tables()
+    rankees_init()
+
+    bot = EloBot(
         slack_client,
         slack_client.get_channel_id(config['channel']),
         config['bot_name'],
         config['min_streak_length'],
     )
+
+    while True:
+        try:
+            bot.run()
+        except KeyboardInterrupt:
+            break
+        except:
+            print(traceback.format_exc())
```

Comment:
include the import here

Reply:
You sure? Convention dictates that all imports be done at the top of the file, as does PEP8: "Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.".

The only reason I can think to do this is to keep memory slightly lower until an exception actually occurs.

Reply:
I know what the convention is, I like the import here because `traceback` is used nowhere else in the file